pandas/plotting/_matplotlib/boxplot.py

```diff
@@ -227,7 +227,8 @@ def _grouped_plot_by_column(
 
     byline = by[0] if len(by) == 1 else by
     fig.suptitle(f"Boxplot grouped by {byline}")
-    fig.subplots_adjust(bottom=0.15, top=0.9, left=0.1, right=0.9, wspace=0.2)
+    if not hasattr(fig, "get_constrained_layout") or not fig.get_constrained_layout():
```

Comment:
make a function for this
maybe_adjust_figure or similar

Reply:
Done. Hope this is what you had in mind.